openeogeotrellis/job_tracker_v2.py

```diff
@@ -85,48 +87,112 @@ class YarnAppReportParseException(Exception):
     """YARN app status getter"""
 
+    def get_application_url(self, application_id: str) -> str:
+        return url_join(
+            ConfigParams().yarn_rest_api_base_url,
+            f"/ws/v1/cluster/apps/{application_id}",
+        )
+
+    def get_authentication_provider(self):
+        auth = None
+        if ConfigParams().yarn_auth_use_kerberos:
+            # TODO: HTTPKerberosAuth is deprecated, this is a utility class for compatibility with requests_kerberos.
+            #   Find the better way to do this, using requests_gssapi directly.
+            # Have to set mutual_authentication to REQUIRED or it just skips kerberos auth it seems.
+            auth = requests_gssapi.HTTPKerberosAuth(
+                mutual_authentication=requests_gssapi.REQUIRED
+            )
+        return auth
+
     def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetadata:
-        try:
-            command = ["yarn", "application", "-status", app_id]
-            with TimingLogger(f"Running {command}", logger=_log.debug):
-                application_report = subprocess.check_output(
-                    command, stderr=subprocess.PIPE
-                ).decode("utf-8")
-        except CalledProcessError as e:
-            stdout = e.stdout.decode()
-            if "doesn't exist in RM or Timeline Server" in stdout:
-                raise AppNotFound(stdout) from None
-            else:
-                raise
-        return self.parse_application_report(application_report)
+        # TODO: Are there any cases were we *do* need to provide credentials to the Kerberos authentication?
+        #   At present we are assuming there is already a Kerberos ticket because kinit was run before
+        #   the geopyspark driver process was started.
+        #   AFAIK that means we don't have to provide user & pw which is in fact the whole point.
+
+        url = self.get_application_url(application_id=app_id)
+        response = requests.get(url, auth=self.get_authentication_provider())
+
+        # Handle error responses: We need the field "message" from "RemoteException"
+        # See https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/WebServicesIntro.html
+        #   specifically, the section: Response Errors
+        #
+        # Example of an error responses body:
+        # {
+        #     "RemoteException": {
+        #         "exception": "NotFoundException",
+        #         "message": "java.lang.Exception: app with id: application_1674538064532_46723 not found",
+        #         "javaClassName": "org.apache.hadoop.yarn.webapp.NotFoundException"
+        #     }
+        # }
+        if response.status_code == 404:
+            remote_exc = response.json().get("RemoteException", {})
+            message = remote_exc.get(
+                "message",
+                "Unknown error. Error response did not contain the message property.",
+            )
+            raise AppNotFound(message) from None
```

Comment:
Can we just pass the entire response as text to `AppNotFound`? We already established that the application does not exist (404) and this seems to just remove context for no clear reason.

Reply:
Done.